src/app/components/itinerary-form/itinerary-form.component.html

```diff
@@ -0,0 +1,140 @@
+<div class="card card-compact mb-2 w-full bg-gray-300 shadow-xl md:w-96">
+  <div class="card-body">
+    <h2 class="card-title">Create an Activity!</h2>
+    <button
+      type="button"
+      class="absolute right-4 top-4 flex items-center gap-1 text-blue-700"
+      onclick="tagModal.showModal()">
+      @if (activityForm.value.tag && activityForm.value.tag !== '') {
+        {{ activityForm.value.tag }}
+        <mat-icon fontIcon="edit"></mat-icon>
+      } @else {
+        Add tag
+        <mat-icon fontIcon="add_circle"></mat-icon>
+      }
+    </button>
+    <div [formGroup]="activityForm">
+      <dialog id="tagModal" class="modal">
+        <div class="modal-box">
+          <form method="dialog">
+            <button
+              type="button"
+              (click)="cancelTag()"
+              class="btn btn-circle btn-ghost btn-sm absolute right-4 top-4">
+              ✕
+            </button>
+          </form>
+          <h3 class="text-lg font-bold">Create Tag!</h3>
+          <mat-form-field hideRequiredMarker="true" class="mt-2 w-full">
+            <mat-label>Tag:</mat-label>
+            <input
+              type="text"
+              matInput
+              placeholder="Type Here..."
+              formControlName="tag" />
+            <mat-error>Please enter a tag</mat-error>
+          </mat-form-field>
+          <p class="py-4">Tap on ✕ to cancel.</p>
+          <form method="dialog" class="flex justify-end">
+            <button
+              type="button"
+              class="btn btn-success btn-sm"
+              onclick="tagModal.close()">
+              Done
+            </button>
+          </form>
+        </div>
+      </dialog>
+      <mat-form-field hideRequiredMarker="true" class="w-full">
+        <mat-label>Title:</mat-label>
+        <input
+          type="text"
+          matInput
+          placeholder="Type Here..."
+          [value]="activityForm.value.title"
+          formControlName="title" />
+        <mat-error>Please enter a valid title</mat-error>
+      </mat-form-field>
+
+      <mat-form-field hideRequiredMarker="true" class="w-full">
+        <mat-label>Enter a date range</mat-label>
+        <mat-date-range-input [dateFilter]="filter" [rangePicker]="picker">
+          <input
+            matStartDate
+            [value]="activityForm.value.startDate"
+            formControlName="startDate"
+            placeholder="Start date" />
+          <input
+            matEndDate
+            [value]="activityForm.value.endDate"
+            formControlName="endDate"
+            placeholder="End date" />
+        </mat-date-range-input>
+        <mat-datepicker-toggle
+          matIconSuffix
+          [for]="picker"></mat-datepicker-toggle>
+        <mat-date-range-picker #picker></mat-date-range-picker>
+        <mat-error>Please enter a valid date range</mat-error>
+      </mat-form-field>
+
+      <div class="mb-4 flex flex-col">
```

Comment:
Don't show it if you don't have it - only submit functional code to the PR please